azure/errors_test.go

```diff
@@ -0,0 +1,41 @@
+package azure
+
+import (
+	"context"
+	"testing"
+	"time"
+)
+
+func TestIsContextDeadlineExceededOrCanceled(t *testing.T) {
+	tests := []struct {
+		name string
+		want bool
+		err  error
+	}{
+		{
+			name: "Context deadline exceeded error",
```

Comment:
can you please add a test case for nil err and one for another type error that returns false?